demo/app/demo/stripe.service.ts

```diff
@@ -36,48 +36,78 @@ export class StripeService implements StripeBackendAPI {
     this.customerSession = new StripeCustomerSession();
   }
 
-  private backendURL(pathComponent: string): string {
-    if (!backendBaseURL) throw new Error("backendBaseURL must be set");
-    if (!backendBaseURL.endsWith("/")) {
-      return backendBaseURL + "/" + pathComponent;
-    } else {
-      return backendBaseURL + pathComponent;
-    }
+  // SetupIntent
+  createSetupIntent(): Promise<any> {
+    return this._postRequest("create_setup_intent").then(response => response.content.toJSON());
+  }
+
+  // PaymentIntent
+  createPaymentIntent(amount: number, currency: string = 'usd'): Promise<any> {
+    const content = `amount=${amount}&currency=${currency}`;
+    return this._postRequest("create_intent", content).then(response => response.content.toJSON());
   }
 
   createCustomerKey(apiVersion: string): Promise<any> {
-    let url = this.backendURL("ephemeral_keys");
-    return httpModule.request({
-      url: url,
-      method: "POST",
-      headers: { "Content-Type": "application/x-www-form-urlencoded; charset=utf-8" },
-      content: "api_version=" + apiVersion
-    }).then(response => {
-      if (response.statusCode < 200 || response.statusCode >= 300) {
-        throw new Error(response.content.toString());
-      }
-      return response.content.toJSON();
-    });
+    const content = `api_version=${apiVersion}`;
+    return this._postRequest("ephemeral_keys", content).then(response => response.content.toJSON());
+  }
+
+  capturePayment(stripeID: string, amount: number, shippingMethod?: StripeShippingMethod, shippingAddress?: StripeAddress): Promise<any> {
+    let content = `payment_method=${stripeID}&amount=${amount}`;
+    if (shippingMethod && shippingAddress) content += `&${this._encodeShipping(shippingMethod, shippingAddress)}`;
+    return this._postRequest("capture_payment", content).then(response => response.content.toJSON());
+  }
+
+  confirmPaymentIntent(paymentIntentID: string): Promise<void> {
+    const content = `payment_intent_id=${paymentIntentID}`;
+    return this._postRequest("confirm_payment", content);
+  }
+
+  createPaymentSession(page: Page, price: number, listener?: StripePaymentListener): StripePaymentSession {
+    return new StripePaymentSession(page, this.customerSession, price, "usd", listener);
+  }
+
+  showPaymentMethods(paymentSession: StripePaymentSession) {
+    paymentSession.presentPaymentMethods();
+  }
+
+  showShipping(paymentSession: StripePaymentSession) {
+    paymentSession.presentShipping();
+  }
+
+  requestPayment(paymentSession: StripePaymentSession) {
+    paymentSession.requestPayment();
   }
 
-  completeCharge(stripeID: string, amount: number, shippingMethod: StripeShippingMethod, shippingAddress: StripeAddress): Promise<void> {
-    let url = this.backendURL("capture_payment");
+  /*
+   *  Private
+   */
+
+  private _postRequest(endpoint: string, content: string = ''): Promise<any> {
```

Comment:
Trivial: For consistency, should probably use " instead of '.

When I started this plugin, Angular was recommending double quotes. Looks like they've now switched to single quotes. I should, too, I guess... sometime.

Reply:
Yup, Only using `'` on my projects, try to stick to the author conventions when doing some PR but old habits die hard 😂